### PR TITLE
Improve handling on obsolete S3 option "force_uri_lib_config"

### PR DESCRIPTION
**Pull request by phoebusm**
#### Reference Issues/PRs
https://github.com/man-group/ArcticDB/issues/832

#### What does this implement/fix? How does it work (high level)? Highlight notable design decisions.
Add `force_uri_lib_config=True` support for backward compatibility

Before https://github.com/man-group/ArcticDB/issues/802:
- No `force_uri_lib_config`: accept
- `force_uri_lib_config=true`: accept
- `force_uri_lib_config=false`: accept

After https://github.com/man-group/ArcticDB/issues/802:
- No `force_uri_lib_config`: accept
- `force_uri_lib_config=true`: raise exception
- `force_uri_lib_config=false`: raise exception

With this PR:
- No `force_uri_lib_config`: accept
- `force_uri_lib_config=true`: accept
- `force_uri_lib_config=false`: raise exception

#### Checklist

<details>
  <summary>
   Checklist for code changes...
  </summary>
 
 - [x] Have you updated the relevant docstrings and documentation?
 - [x] Is this contribution tested against [all ArcticDB's features](../docs/mkdocs/docs/technical/contributing.md)?
 - [x] Do all exceptions introduced raise appropriate [error messages](https://docs.arcticdb.io/error_messages/)?
 - [x] Are API changes highlighted in the PR description?
 - [x] Is the PR labelled as enhancement or bug so it appears in autogenerated release notes?
</details>

### CODE REVIEW

**Review by jamesmunro — APPROVED** [review]

File: python/arcticdb/adapters/s3_library_adapter.py [hunk 1]
```diff
@@ -55,6 +57,12 @@ def __init__(self, uri: str, encoding_version: EncodingVersion, *args, **kwargs)
 
         self._query_params: ParsedQuery = self._parse_query(match["query"])
 
+        if self._query_params.force_uri_lib_config == False:
```
Comment (mehertz) [inline python/arcticdb/adapters/s3_library_adapter.py:60]: `is False` is more Pythonic

File: python/arcticdb/adapters/s3_library_adapter.py [hunk 2]
```diff
@@ -38,6 +38,8 @@ class ParsedQuery:
 
     path_prefix: Optional[str] = None
 
+    force_uri_lib_config: Optional[bool] = True
```
[the remark this answers is not in the document]
Comment (phoebusm) [reply python/arcticdb/adapters/s3_library_adapter.py:41]: Hmmm I think my exception message below should have explained this message in the exception message below, with the support link to the more detailed one as well

**Comment by mehertz** [comment]
Presumably this option has already been removed from the documentation?

File: python/arcticdb/adapters/s3_library_adapter.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mehertz) [reply python/arcticdb/adapters/s3_library_adapter.py:41]: There's a very good chance these options get refactored to a different file very separate from your exception message below at some point. As a result I think it's worth having a comment line inline in the config so a copy/paste of this class keeps the distinction.

**Review by mehertz — APPROVED** [review]

File: python/arcticdb/adapters/s3_library_adapter.py (hunk 1 above)
Comment (phoebusm) [reply python/arcticdb/adapters/s3_library_adapter.py:60]: Will do

File: python/arcticdb/adapters/s3_library_adapter.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mehertz) [reply python/arcticdb/adapters/s3_library_adapter.py:41]: FWIW I literally just think one line is enough:
```
# DEPRECATED - see https://github.com/man-group/ArcticDB/pull/833
force_uri_lib_config: Optional[bool] = True
```

**Pull request merged**